iceoryx_utils/include/iceoryx_utils/internal/cxx/vector.inl

```diff
@@ -201,7 +201,7 @@ inline const T& vector<T, Capacity>::at(const uint64_t index) const
 {
     if (index + 1 > m_size)
     {
-        fprintf(stderr, "out of bounds access, current size is %lu but given index is %lu\n", m_size, index);
+        std::cerr << "out of bounds access, current size is " << m_size <<  " but given index is " << index << std::endl;
```

Comment:
@elfenpiff What was the reason for chosing fprintf here?

Reply:
might be nit, but within the ROS2 context we generally preferred `fprintf` over `cerr/cout` as it is primarily threadsafe and doesn't produce interleaved output when running concurrently.

Reply:
The problem with fprintf here is that index is a `uint64_t` and is not guaranteed to be an `unsigned long`. Indeed on my Mac, it's an `unsigned long long` and generates a build warning.

Reply:
the right specifier would be `PRIu64` in this case. See here:
http://www.cplusplus.com/reference/cinttypes/

so in this case:
```c++
fprintf(stderr, "out of bounds access, current size is %lu but given index is %" PRIu64 "\n", m_size, index);
```

Reply:
I’d rather use cerr to be consistent with the surrounding code

Reply:
That's true. Everywhere else in this file std::cerr is used. So it would just make it consistent. But I'm not sure if std::cerr and std::terminate should be our default error handling here.